oura-library/src/main/kotlin/org/radarbase/oura/request/OuraRequestGenerator.kt

```diff
@@ -145,11 +145,13 @@ constructor(
                     ?: nextRequestTime
         } else {
             if (request.startDate.plus(TIME_AFTER_REQUEST).isBefore(Instant.now())) {
+                logger.info("No records found, updating offsets to end date..")
                 ouraOffsetManager.updateOffsets(
                     request.route,
                     request.user,
                     request.endDate,
                 )
+                userNextRequest[request.user.versionedId] = Instant.now().plus(ONE_DAY)
```

Comment:
@yatharthranjan Should we update this to less than one day? I just realised if the duration between now and the start date is long, then it would take a while to pull all the data? E.g. for historical data that starts 2 years prior.